src/linear_algebra/tensor_algebra.lean

```diff
@@ -63,6 +63,11 @@ namespace tensor_algebra
 instance {S : Type*} [comm_ring S] [semimodule S M] : ring (tensor_algebra S M) :=
 ring_quot.ring _
 
+/-- Once we mark tensor_algebra irreducible, we can't prove this any more -/
+@[simp]
+lemma semiring_eq_ring {S : Type*} [comm_ring S] [semimodule S M] :
+  ring.to_semiring = tensor_algebra.semiring S M := rfl
```

Comment:
Needing this lemma suggests we will need a lot of `simp`ing when mixing `semiring` and `ring`. I think the correct approach is to write something like `instance : ring (tensor_algebra S M) := { .. tensor_algebra.semiring S M, .. ring_quot.ring _ }` (or maybe in the other order, I don't recall which ones have priority), then the projection to `semiring` should be defeq even if `tensor_algebra` is irreducible.

Reply:
Neither order helps unfortunately - this lemma still can't be proved once the `irreducible` attribute is added.